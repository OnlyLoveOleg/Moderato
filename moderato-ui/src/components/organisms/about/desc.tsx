import { NextComponentType, NextPageContext } from 'next';
import styled from 'styled-components';
import { H2, H3 } from '@/components/atoms/heading';
import { posUppLeft, MediaSP } from '@/styles/styled-components/global';

const AboutDesc = styled.div`
  color: ${(props) => props.theme.white};
  background-color: #222222e6;
  user-select: none;
  position: relative;

  > .desc-title {
    ${posUppLeft({ top: '-6%', left: '0' })};
    font-family: 'Mrs Saint Delafield', cursive;
    transform: rotate(-15deg);
  }

  > .desc-subtitle {
    padding: 2rem 1rem;
  }

  > .desc {
    font-size: 0.75rem;
    color: ${(props) => props.theme.descColor};
  }

  > .ja {
    padding: 0 1rem 1rem;
  }

  > .en {
    padding: 0 1rem 1rem;
  }

  ${MediaSP`
    .desc-subtitle {
      font-size: 1.5rem;
    }

    .desc {
      font-size: 0.8rem;
    }

    .en {
      padding: 0 1rem 100vh;
    }
  `}
`;

// どこかで管理する？

export const Desc: NextComponentType<NextPageContext> = () => {
  return (
    <AboutDesc date-testid='about-desc'>
      <H2 className='desc-title' size='2rem' text='Coding in Design' />
      <H3
        className='desc-subtitle'
        size='1rem'
        text='My mission is to write coding as beautifully as design and to make the whole team excited!'
      />
      <p className='desc ja'>
        千葉出身。都内（銀座）での美容師経験があるソフトウェアエンジニア。モデル撮影のために購入したmacを使っていたらPCの仕組みが気になり始め、気づいたらエンジニアに。人生まるごとクリエイティブだと思っている。人の髪をデザインしていた経験をIT業界へ活かせることはないか日々模索中。
      </p>
      <p className='desc en'>
        Born in Japan. Software engineer with experience as a beautician in Tokyo (Ginza). When he
        was using a mac he bought for a model shoot, he started to be interested in how PCs work,
        and he found himself becoming an engineer. He believes that one&apos;s entire life is
        creative. He is always looking for ways to apply his experience in designing people&apos;s
        hair to the IT industry.
      </p>
    </AboutDesc>
  );
};
