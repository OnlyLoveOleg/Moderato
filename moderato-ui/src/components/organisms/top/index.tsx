export * from './aboutSection';
export * from './recommendSection';
export * from './horizontalScroll';
