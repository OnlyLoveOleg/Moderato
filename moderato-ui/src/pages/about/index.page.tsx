import type { NextPage } from 'next';
import { About as AbTpl, Layout } from '@/components/templates';

const About: NextPage = () => {
  return (
    <Layout disableRightClick={true}>
      <AbTpl />
    </Layout>
  );
};

export default About;
