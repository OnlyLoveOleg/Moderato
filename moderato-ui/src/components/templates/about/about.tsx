import { NextComponentType, NextPageContext } from 'next';
import Image from 'next/image';
import { Desc } from '@/components/organisms/about';
import styled from 'styled-components';
import { displayFlex, posUppLeft, MediaSP, posUppRight } from '@/styles/styled-components/global';

const Wrapper = styled.div`
  ${displayFlex({ flexDirection: 'row' })};
  height: 100vh;
  width: 100%;
  background-color: ${(props) => props.theme.black};

  /* > 直下 */
  > .image-wrap {
    ${posUppLeft({ top: '20%', left: '10%' })};
    width: 45%;
  }

  > .about-wrap {
    ${posUppRight({ top: '25%', right: '10%' })};
    width: 45%;
  }

  // position を absolute や fixed していたものを解除するには position: initial; で上書きしてリセットします
  ${MediaSP`
    ${displayFlex({ flexDirection: 'column' })};

    .image-wrap, .about-wrap {
      position: initial;
      width: 100%;
      height: 50%;
    }
  `}
`;

// next/image see https://www.wantedly.com/companies/wantedly/post_articles/385515
export const About: NextComponentType<NextPageContext> = () => {
  return (
    <Wrapper data-testid='about-tpl'>
      <div className='image-wrap'>
        <Image
          src='/images/about_me.jpg'
          width='700'
          height='525'
          alt='me_image'
          objectFit='contain'
          onError={(e) => {
            e.currentTarget.src = `https://placehold.jp/32/003060/e0e0e0/300x200.png`;
          }}
        />
      </div>
      <div className='about-wrap'>
        <Desc />
      </div>
    </Wrapper>
  );
};
