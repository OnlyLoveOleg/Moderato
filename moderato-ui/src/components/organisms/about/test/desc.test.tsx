import { render, screen } from '@testing-library/react';
import { Desc } from '@/components/organisms/about';
import { screenShot } from '@/tests/util';

describe('', () => {
  test('Check About Desc view', () => {
    render(<Desc />); // これでscreenから要素を取得できる。

    expect(screen.getByText('Coding in Design')).toBeInTheDocument();
    expect(
      screen.getByText(
        'My mission is to write coding as beautifully as design and to make the whole team excited!',
      ),
    ).toBeInTheDocument();
    // TODO GSAP対応は今度する
    // expect(
    //   screen.getByText(
    //     '千葉出身。都内（銀座）での美容師経験があるソフトウェアエンジニア。モデル撮影のために購入したmacを使っていたらPCの仕組みが気になり始め、気づいたらエンジニアに。人生まるごとクリエイティブだと思っている。人の髪をデザインしていた経験をIT業界へ活かせることはないか日々模索中。',
    //   ),
    // ).toBeInTheDocument();
  });

  test('Check snapshot', () => {
    const tree = screenShot(<Desc />);
    expect(tree).toMatchSnapshot();
  });
});
