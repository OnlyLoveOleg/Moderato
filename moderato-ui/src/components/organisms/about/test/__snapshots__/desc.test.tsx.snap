// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Check snapshot 1`] = `
<div
  className="sc-e80d70df-0 kXiClh"
>
  <h2
    className="sc-1ef67e09-0 bWTTkQ desc-title"
    size="2rem"
  >
    Coding in Design
  </h2>
  <h3
    className="sc-31ba8a03-0 iygQXX desc-subtitle"
    size="1rem"
  >
    My mission is to write coding as beautifully as design and to make the whole team excited!
  </h3>
  <p
    className="sc-21bc2e1b-0 dyIjIH desc ja"
    id="souseki"
  >
    千葉出身。都内（銀座）での美容師経験があるウェブエンジニア。モデル撮影のために購入したmacを使っていたらPCの仕組みが気になり始め、気づいたらエンジニアに。人生まるごとクリエイティブだと思っている。人の髪をデザインしていた経験をIT業界へ活かせることはないか日々模索中。
  </p>
  <p
    className="sc-21bc2e1b-0 dyIjIH desc en"
    id="souseki2"
  >
    Born in Japan. Software engineer with experience as a beautician in Tokyo (Ginza). When he was using a mac he bought for a model shoot, he started to be interested in how PCs work, and he found himself becoming an engineer. He believes that one's entire life is creative. He is always looking for ways to apply his experience in designing people's hair to the IT industry.
  </p>
</div>
`;
